File: js/src/maplestory/api/tms/dto/character/characterPopularity.ts
import * as base from '../../../common/dto/character/characterPopularity';
import { CharacterPopularityBody } from '../../response/character/characterPopularityBody';

/**
 * 角色名聲資訊
 */
export class CharacterPopularityDto extends base.CharacterPopularityDto {
  /**
   * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
   */
  public override date: Date | null;

  /**
   * 角色名聲
   */
  public override popularity: number;

  constructor(obj: CharacterPopularityBody) {
    super();

    const { date, popularity } = obj;

    this.date = date ? new Date(date) : null;
    this.popularity = popularity!;
  }
}

File: python/maplestory_openapi/api/tms/dto/character/character_popularity.py
from datetime import datetime
from pydantic import BaseModel

from maplestory_openapi.api.common.dto.character.character_popularity import CharacterPopularity as BaseCharacterPopularity


class CharacterPopularity(BaseModel, BaseCharacterPopularity):
    """
    角色名聲資訊

    Attributes:
        date (datetime or None): 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
        popularity (int): 角色名聲
    """
    date: datetime | None
    popularity: int
